### ISSUE TITLE
Custom script execution failing

### ISSUE THREAD

**Issue by dsiluri** [elastic/azure-marketplace#308]
Hi,
following template deployment throws an exception while executing a custom script extension while provisioning a VM, are you able to tell me what is trying to connect to please?

Enable failed: processing file downloads failed: failed to download file[0]: failed to download file: http request failed: Get [REDACTED] dial tcp 151.101.208.133:443

Command executed:
$template_base_uri="https://raw.githubusercontent.com/elastic/azure-marketplace"
$template_version="7.3.1"

az group deployment create --resource-group "dev-elk"  --template-uri $template_base_uri/$template_version/src/mainTemplate.json  --parameters _artifactsLocation=$template_base_uri/$template_version/src/ esVersion=7.3.1 esClusterName=elasticsearch vmDataDiskCount=1 dataNodesAreMasterEligible=Yes adminUsername=kaiser adminPassword=********** securityBootstrapPassword=*********securityAdminPassword=******** securityKibanaPassword=******** securityLogstashPassword=********** securityBeatsPassword=****** securityApmPassword=********* securityRemoteMonitoringPassword=*********** vmDataNodeCount=2 xpackPlugins=No logstash=Yes vmLogstashCount=2 loadBalancerType=internal loadBalancerInternalSku=Standard --vNetName elk-vnet --debug

**Comment by russcam** [comment]
Can you provide more detail @dsiluri please? Some details that would be useful

- The error message / log file from the failed deployment
- [Logs from the VMs](https://www.elastic.co/guide/en/elastic-stack-deploy/current/azure-arm-template-troubleshooting-azure-logs.html)

**Comment by RomasZekonis** [comment]
Hi 
I'm receiving the same error when deploying with script. Only on data nodes. tried a lot deployments-redeployment. Logged in to data node - looks that VM do not have internet access. That why script execution is failing on data nodes. Logstash and Kibana nodes do have access, data nodes don't.  

The cause of this issue is "loadBalancerInternalSku". If  "Standard" then fails if "Basic" then passes.
Different tier load balancer differently routes outgoing traffic from nodes.

There is a nice comparison of load balancers tiers https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-overview#skus

To make it working with Standard tier maybe require to create some outbound rules.

**Comment by russcam** [comment]
> The cause of this issue is "loadBalancerInternalSku". If "Standard" then fails if "Basic" then passes.
Different tier load balancer differently routes outgoing traffic from nodes.

There's an integration test that deploys a cluster with `Standard` SKU internal and external load balancers. It can be run with

```sh
npm run release -- --test 1d-0m-1c-ext-slb-kp
```
If you want to keep the resource group around after the tests have run, add the `--nodestroy` flag to the end.

The test passes successfully:
```
[14:56:13] Working directory changed to ~\source\azure-marketplace\build
[14:56:14] Using gulpfile ~\source\azure-marketplace\build\gulpfile.js
[14:56:14] Starting 'release'...
[14:56:14] Starting 'default'...
[14:56:14] Starting 'sanity-checks'...
[14:56:14] Finished 'sanity-checks' after 20 ms
[14:56:14] Starting 'patch'...
[14:56:14] Finished 'patch' after 16 ms
[14:56:14] Starting '<anonymous>'...
[14:56:14] Finished '<anonymous>' after 261 ms
[14:56:14] Finished 'default' after 301 ms
[14:56:14] Starting 'deploy'...
[14:56:14] Starting 'clean'...
[14:56:14] Starting 'create-log-folder'...
[14:56:14] Finished 'create-log-folder' after 1.84 ms
[14:56:14] Starting '<anonymous>'...
[14:56:14] Finished '<anonymous>' after 1.84 ms
[14:56:14] Finished 'clean' after 6.57 ms
[14:56:14] Starting '<anonymous>'...
[14:56:14] Using version 7.5.0 for tests
[14:56:14] Using template: https://raw.githubusercontent.com/elastic/azure-marketplace/7.5/src/mainTemplate.json
[14:56:18] Using azure-cli 2.0.77
[14:56:18] logging into azure cli tooling
[14:56:21] start validating 1 arm test scenarios
[14:56:21] creating resource group: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST
[14:56:29] validating 1d-0m-1c-ext-slb-kp in resource group: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST
[14:56:40] all 1 arm test scenarios validated
[14:56:40] start deploying 1 valid arm test scenarios
[14:56:40] deploying 1d-0m-1c-ext-slb-kp in resource group: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST
[15:05:20] checking external loadbalancer http://lb-<random>.westeurope.cloudapp.azure.com:9200 in resource group: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST
[15:05:20] checking kibana at http://kb-<random>.westeurope.cloudapp.azure.com:5601 in resource group: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST
[15:05:21] [PASS] expecting license in resource group test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST to be: trial and found: trial
[15:05:21] [PASS] kibana is running in resource group: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST with state: green
[15:05:21] checking kibana monitoring endpoint for rg: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST
[15:05:22] [PASS] cluster is up and running in resource group: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST
[15:05:22] [PASS] expecting 2 total nodes in resource group: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST and found: 2
[15:05:22] [PASS] expecting 1 data nodes in resource group: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST and found: 1
[15:05:22] all 1 valid arm test scenarios deployed
[15:05:22] deleting 1 resource groups
[15:05:22] deleting resource group: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST
[15:05:29] deleted resource group: test-xtreme-1d-0m-1c-ext-slb-kp-20191211-145614-EST
[15:05:29] logging out of azure cli tooling
[15:05:30] ============================================
[15:05:30] Integration test results
[15:05:30] ============================================
[15:05:30] PASS : 5
[15:05:30] FAIL : 0
[15:05:30] TOTAL: 5
[15:05:30] ============================================
[15:05:30] deleting temporary parameter files
[15:05:30] Finished '<anonymous>' after 9.25 min
[15:05:30] Finished 'deploy' after 9.25 min
[15:05:30] Starting '<anonymous>'...
[15:05:30] Finished '<anonymous>' after 16 ms
[15:05:30] Finished 'release' after 9.27 min
```

and ssh'ing into the data node, it has internet access. I've infrequently seen intermittent issues with Azure DNS, whereby resolution fails to resolve the artifacts URIs to download packages to bootstrap the cluster, but it doesn't happen too often. 

> To make it working with Standard tier maybe require to create some outbound rules.

There's an NSG configured to allow traffic from the external LB to the VMs when the Standard SKU is used, but this isn't needed for the internal LB according to the docs.

We’re closing this issue due to inactivity; if you’re still in need of resolution, feel free to re-open with the additional details and we'd be glad to help you out 👍

**Comment by russcam** [comment]
My mistake, I can replicate it when _only_ a `Standard` internal LB is deployed and `loadBalancerType` is internal. Will look to fix this for the next release, 7.5.0
